pkg/provision/linode.go

```diff
@@ -0,0 +1,140 @@
+package provision
+
+import (
+	"context"
+	"fmt"
+	"github.com/linode/linodego"
+	"github.com/sethvargo/go-password/password"
+	"golang.org/x/oauth2"
+	"net/http"
+	"strconv"
+)
+
+type LinodeInterface interface {
+	CreateStackscript(createOpts linodego.StackscriptCreateOptions) (*linodego.Stackscript, error)
+	CreateInstance(instance linodego.InstanceCreateOptions) (*linodego.Instance, error)
+	GetInstance(linodeID int) (*linodego.Instance, error)
+	DeleteInstance(linodeID int) error
+	DeleteStackscript(id int) error
+}
+
+type LinodeClient struct {
+	client linodego.Client
+}
+
+func (p *LinodeClient) CreateStackscript(createOpts linodego.StackscriptCreateOptions) (*linodego.Stackscript, error) {
+	return p.client.CreateStackscript(context.Background(), createOpts)
+}
+
+func (p *LinodeClient) CreateInstance(instance linodego.InstanceCreateOptions) (*linodego.Instance, error) {
+	return p.client.CreateInstance(context.Background(), instance)
+}
+
+func (p *LinodeClient) GetInstance(linodeID int) (*linodego.Instance, error) {
+	return p.client.GetInstance(context.Background(), linodeID)
+}
+
+func (p *LinodeClient) DeleteInstance(linodeID int) error {
+	return p.client.DeleteInstance(context.Background(), linodeID)
+}
+
+func (p *LinodeClient) DeleteStackscript(id int) error {
+	return p.client.DeleteStackscript(context.Background(), id)
+}
+
+type LinodeProvisioner struct {
+	client        LinodeInterface
+	stackscriptID int
+}
+
+func NewLinodeProvisioner(apiKey string) (*LinodeProvisioner, error) {
+	tokenSource := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: apiKey})
+
+	oauth2Client := &http.Client{
+		Transport: &oauth2.Transport{
+			Source: tokenSource,
+		},
+	}
+
+	linodeClient := linodego.NewClient(oauth2Client)
+	return &LinodeProvisioner{
+		client: &LinodeClient{linodeClient},
+	}, nil
+}
+
+// Provision provisions a new Linode instance as an exit node
+func (p *LinodeProvisioner) Provision(host BasicHost) (*ProvisionedHost, error) {
+
+	// Create a stack script with the inlets user data as contents.
+	// For security concern, this script is created as IsPublic: false.
```

Comment:
"For security concern" <- For which "security concern"? Can you explain in the comment please?

What is a "Stackscript"?

Reply:
Updated the comments with detailed explanation.